docs/docsite/rst/user_guide/playbooks_tags.rst

```diff
@@ -1,227 +1,402 @@
 
-    - debug:
-        msg: "runs when you use tag1"
-      tags:
-      - tag1
+Previewing the results of using tags
+------------------------------------
 
-.. versionadded:: 2.5
+Especially when you run a role or playbook someone else has written or edited, you may not know which tasks have which tags, or which tags exist at all. Ansible offers two command-line flags for :ref:`ansible-playbook` that help you manage tagged playbooks:
+
+* ``--list-tags`` - generate a list of available tags
+* ``--list-tasks`` - when used with ``--tags tagname`` or ``--skip-tags tagname``, generate a preview of tagged tasks
+
+For example, if you do not know whether the tag for configuration tasks is ``config`` or ``conf`` in a playbook, role, or tasks file, you can display all available tags without running any tasks:
+
+.. code-block:: bash
+
+   ansible-playbook example.yml --list-tags
+
+If you do not know which tasks have the tags ``configuration`` and ``packages``, you can pass those tags and add ``--list-tasks``. Ansible lists the tasks but does not execute any of them.
+
+.. code-block:: bash
+
+   ansible-playbook example.yml --tags "configuration,packages" --list-tasks
+
+These command-line flags have one limitation - they cannot show tags or tasks within dynamically included files or roles. See :ref:`dynamic_vs_static` for more information on differences between static imports and dynamic includes.
+
+.. _selective_reuse:
+
+Selectively running tagged tasks in re-usable files
+---------------------------------------------------
+
+If you have a role or a tasks file with tags defined at the task or block level, you can selectively run or skip those tagged tasks in a playbook if you use a dynamic include instead of a static import. You must use the same tag on the included tasks and on the include statement itself. For example if you use a file like this:
+
+.. code-block:: yaml
+
+   # mixed.yml
+   tasks:
+   - name: task with no tags
+     debug:
+       msg: this task has no tags
+
+   - name: tagged task
+     debug:
+       msg: this task is tagged with mytag
+     tags: mytag
 
-Another special tag is ``never``, which will prevent a task from running unless
-a tag is specifically requested.
+   - block:
+     - name: First block task with mytag
+       ...
+     - name: Second block task with mytag
+       ...
+     tags:
+     - mytag
 
-Example::
+in a playbook like this:
 
-    tasks:
-      - debug: msg="{{ showmevar }}"
-        tags: [ never, debug ]
+.. code-block:: yaml
 
-In this example, the task will only run when the ``debug`` or ``never`` tag
-is explicitly requested.
+   # myplaybook.yml
+   - hosts: all
+     tasks:
+     - include_tasks:
+         name: mixed.yml
+       tags: mytag
 
+and run the playbook with ``ansible-playbook -i hosts myplaybook.yml --tags "mytag"``, Ansible skips the task with no tags, runs the tagged individual task, and runs the two tasks in the block.
```

Comment:
"When you run the playbook with ``ansible-playbook -i hosts myplaybook.yml --tags "mytag"``, Ansible skips the..."